pkgs/by-name/ni/nixos-render-docs/src/nixos_render_docs/manual.py

```diff
@@ -44,14 +45,18 @@ def convert(self, infile: Path, outfile: Path) -> None:
     def _postprocess(self, infile: Path, outfile: Path, tokens: Sequence[Token]) -> None:
         pass
 
-    def _handle_headings(self, tokens: list[Token], *, on_heading: Callable[[Token,str],None]) -> None:
+    def _handle_headings(self, tokens: list[Token], *, src: str, on_heading: Callable[[Token,str],None]) -> None:
```

Comment:
Most of the diff is like this: passing the `src` around so we can access it in the exceptions and replacing `RuntimeError` with `SrcError`.

If you'd like, I can split it so that this PR only replaces one `raise` and move the rest of these changes to a subsequent PR.